sky/execution.py

```diff
@@ -516,19 +517,38 @@ def spot_launch(
         common_utils.dump_yaml(f.name, task_config)
 
         controller_name = spot.SPOT_CONTROLLER_NAME
+        vars_to_fill = {
+            'remote_user_yaml_prefix': spot.SPOT_TASK_YAML_PREFIX,
+            'user_yaml_path': f.name,
+            'user_config_path': None,
+            'spot_controller': controller_name,
+            'cluster_name': name,
+            'gcloud_installation_commands': gcp.GCLOUD_INSTALLATION_COMMAND,
+            'is_dev': env_options.Options.IS_DEVELOPER.get(),
+            'disable_logging': env_options.Options.DISABLE_LOGGING.get(),
+            'logging_user_hash': common_utils.get_user_hash(),
+            'retry_until_up': retry_until_up,
+            'user': os.environ.get('USER', None),
+        }
+        if sky_config.loaded():
+            # Look up the contents of the already loaded configs via the
+            # 'sky_config' module. Don't simply read the on-disk file as it may
+            # have changed since this process started.
+            #
+            # Pop any proxy command, because the controller would've been
+            # launched behind the proxy, and in general any nodes we launch may
```

Comment:
The spot controller can be launched in a different VPC if the user changed the `config` after their first `sky spot launch`. Do we want to check if the VPC in the `sky_config` matches the one in the current `sky_config`? Maybe we can check the generated ray yaml to find the previous information?

Reply:
Oh, I mean if the user changes the VPC in the `sky_config`, the newly launched spot cluster will be configured to be launched in a different VPC against the controller. Will that cause the controller to fail to access that spot cluster? 
Should we error out when we detect the user-configured VPC changed against the controller when `sky spot launch` is called?

Reply:
Great catch! Added a comment here:
```
            # NOTE: suppose that we have a controller in old VPC, then user
            # changes 'vpc_name' in the config and does a 'spot launch'. In
            # general, the old controller may not successfully launch the job
            # in the new VPC. This happens if the two VPCs don’t have peering
            # set up. Like other places in the code, we assume properly setting
            # up networking is user's responsibilities.
```

Brainstorming a potential safeguard logic:
- If controller has no sky_config file:
  - Assert that on a new spot launch, no sky_config is loaded — i.e., both will be in the same default VPC
If controller has a sky_config file:
- Assert that on a new spot launch, sky_config is loaded, and its aws.vpc_name field is the same name as the controller’s.
  - This may throw some false negatives:
    - e.g., the spot job may be launched in non-AWS
  - This may throw some false positives:
    - e.g., even if the vpc_name is the same, the spot job may happen to be launched in a different region’s VPC with that name, and it may be the case that the 2 VPCs don’t have peering set up.
    
So it seems this check is insufficient. It may, however, strongly remind users that “vpc_name: xxx” is a special VPC in each region & you need to set up peering.

As discussed, we decided to defer this.